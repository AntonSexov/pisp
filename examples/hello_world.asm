.data
    wordtest: "nice"
    skibidi: 5
.code 
    begin:
        load 0 
        load 44
        store &25
        add 44
        sub 99
        mod 5
        store &26
        lbl:
            load &3
            load &4
        load &5
        load 11
        store &7
        store &28
        load 2
        sub 1
        jifnn &14
    end