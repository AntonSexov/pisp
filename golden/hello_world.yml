in_src: |-
  .data
      hello: "Hello, world!"
      tmp: 2
  .code 
      begin:
          load hello
          mainloop:
              store out
              load tmp
              add 1
              store tmp
              load  &tmp        
              jifnz mainloop
              hlt
      end
in_input: |-
  []
out_src: |-
  [
      {
          "idx": 2,
          "opcode": "nop",
          "operand": "H",
          "address": false
      },
      {
          "idx": 3,
          "opcode": "nop",
          "operand": "e",
          "address": false
      },
      {
          "idx": 4,
          "opcode": "nop",
          "operand": "l",
          "address": false
      },
      {
          "idx": 5,
          "opcode": "nop",
          "operand": "l",
          "address": false
      },
      {
          "idx": 6,
          "opcode": "nop",
          "operand": "o",
          "address": false
      },
      {
          "idx": 7,
          "opcode": "nop",
          "operand": ",",
          "address": false
      },
      {
          "idx": 8,
          "opcode": "nop",
          "operand": " ",
          "address": false
      },
      {
          "idx": 9,
          "opcode": "nop",
          "operand": "w",
          "address": false
      },
      {
          "idx": 10,
          "opcode": "nop",
          "operand": "o",
          "address": false
      },
      {
          "idx": 11,
          "opcode": "nop",
          "operand": "r",
          "address": false
      },
      {
          "idx": 12,
          "opcode": "nop",
          "operand": "l",
          "address": false
      },
      {
          "idx": 13,
          "opcode": "nop",
          "operand": "d",
          "address": false
      },
      {
          "idx": 14,
          "opcode": "nop",
          "operand": "!",
          "address": false
      },
      {
          "idx": 15,
          "opcode": "nop",
          "operand": "\u0000",
          "address": false
      },
      {
          "idx": 16,
          "opcode": "nop",
          "operand": 2,
          "address": false
      },
      {
          "idx": 0,
          "opcode": "load",
          "operand": 2,
          "address": true,
          "relative": false
      },
      {
          "idx": 1,
          "opcode": "store",
          "operand": 1,
          "address": true,
          "relative": false
      },
      {
          "idx": 2,
          "opcode": "load",
          "operand": 16,
          "address": true,
          "relative": false
      },
      {
          "idx": 3,
          "opcode": "add",
          "operand": 1,
          "address": false,
          "relative": false
      },
      {
          "idx": 4,
          "opcode": "store",
          "operand": 16,
          "address": true,
          "relative": false
      },
      {
          "idx": 6,
          "opcode": "load",
          "operand": 16,
          "address": true,
          "relative": true
      },
      {
          "idx": 7,
          "opcode": "jifnz",
          "operand": 1,
          "address": true,
          "relative": false
      },
      {
          "idx": 8,
          "opcode": "hlt",
          "operand": null,
          "address": false,
          "relative": false
      }
  ]
out_log: |
    INFO INSTRUCTION: 1 | LOAD: DR => ACC | AR: 2 | IP: 1 | DR: H | ACC: H | NZC: 0 | TICK: 4| CR:{'idx': 0, 'opcode': 'load', 'operand': 2, 'address': True, 'relative': False}
    INFO INSTRUCTION: 2 | STORE: ACC => RAM[AR] | AR: 1 | IP: 2 | DR: H | ACC: H | NZC: 0 | TICK: 8| CR:{'idx': 1, 'opcode': 'store', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 3 | LOAD: DR => ACC | AR: 16 | IP: 3 | DR: 2 | ACC: 2 | NZC: 0 | TICK: 12| CR:{'idx': 2, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 4 | ADD: ACC + DR => ACC | AR: 3 | IP: 4 | DR: 1 | ACC: 3 | NZC: 0 | TICK: 16| CR:{'idx': 3, 'opcode': 'add', 'operand': 1, 'address': False, 'relative': False}
    INFO INSTRUCTION: 5 | STORE: ACC => RAM[AR] | AR: 16 | IP: 5 | DR: 1 | ACC: 3 | NZC: 0 | TICK: 20| CR:{'idx': 4, 'opcode': 'store', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 6 | LOAD: DR => ACC | AR: 16 | IP: 6 | DR: e | ACC: e | NZC: 0 | TICK: 26| CR:{'idx': 6, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': True}
    INFO INSTRUCTION: 7 | JIFNZ:NOT Z: DR => IP | AR: 6 | IP: 1 | DR: 1 | ACC: e | NZC: 0 | TICK: 30| CR:{'idx': 7, 'opcode': 'jifnz', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 8 | STORE: ACC => RAM[AR] | AR: 1 | IP: 2 | DR: 1 | ACC: e | NZC: 0 | TICK: 34| CR:{'idx': 1, 'opcode': 'store', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 9 | LOAD: DR => ACC | AR: 16 | IP: 3 | DR: 3 | ACC: 3 | NZC: 0 | TICK: 38| CR:{'idx': 2, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 10 | ADD: ACC + DR => ACC | AR: 3 | IP: 4 | DR: 1 | ACC: 4 | NZC: 0 | TICK: 42| CR:{'idx': 3, 'opcode': 'add', 'operand': 1, 'address': False, 'relative': False}
    INFO INSTRUCTION: 11 | STORE: ACC => RAM[AR] | AR: 16 | IP: 5 | DR: 1 | ACC: 4 | NZC: 0 | TICK: 46| CR:{'idx': 4, 'opcode': 'store', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 12 | LOAD: DR => ACC | AR: 16 | IP: 6 | DR: l | ACC: l | NZC: 0 | TICK: 52| CR:{'idx': 6, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': True}
    INFO INSTRUCTION: 13 | JIFNZ:NOT Z: DR => IP | AR: 6 | IP: 1 | DR: 1 | ACC: l | NZC: 0 | TICK: 56| CR:{'idx': 7, 'opcode': 'jifnz', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 14 | STORE: ACC => RAM[AR] | AR: 1 | IP: 2 | DR: 1 | ACC: l | NZC: 0 | TICK: 60| CR:{'idx': 1, 'opcode': 'store', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 15 | LOAD: DR => ACC | AR: 16 | IP: 3 | DR: 4 | ACC: 4 | NZC: 0 | TICK: 64| CR:{'idx': 2, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 16 | ADD: ACC + DR => ACC | AR: 3 | IP: 4 | DR: 1 | ACC: 5 | NZC: 0 | TICK: 68| CR:{'idx': 3, 'opcode': 'add', 'operand': 1, 'address': False, 'relative': False}
    INFO INSTRUCTION: 17 | STORE: ACC => RAM[AR] | AR: 16 | IP: 5 | DR: 1 | ACC: 5 | NZC: 0 | TICK: 72| CR:{'idx': 4, 'opcode': 'store', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 18 | LOAD: DR => ACC | AR: 16 | IP: 6 | DR: l | ACC: l | NZC: 0 | TICK: 78| CR:{'idx': 6, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': True}
    INFO INSTRUCTION: 19 | JIFNZ:NOT Z: DR => IP | AR: 6 | IP: 1 | DR: 1 | ACC: l | NZC: 0 | TICK: 82| CR:{'idx': 7, 'opcode': 'jifnz', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 20 | STORE: ACC => RAM[AR] | AR: 1 | IP: 2 | DR: 1 | ACC: l | NZC: 0 | TICK: 86| CR:{'idx': 1, 'opcode': 'store', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 21 | LOAD: DR => ACC | AR: 16 | IP: 3 | DR: 5 | ACC: 5 | NZC: 0 | TICK: 90| CR:{'idx': 2, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 22 | ADD: ACC + DR => ACC | AR: 3 | IP: 4 | DR: 1 | ACC: 6 | NZC: 0 | TICK: 94| CR:{'idx': 3, 'opcode': 'add', 'operand': 1, 'address': False, 'relative': False}
    INFO INSTRUCTION: 23 | STORE: ACC => RAM[AR] | AR: 16 | IP: 5 | DR: 1 | ACC: 6 | NZC: 0 | TICK: 98| CR:{'idx': 4, 'opcode': 'store', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 24 | LOAD: DR => ACC | AR: 16 | IP: 6 | DR: o | ACC: o | NZC: 0 | TICK: 104| CR:{'idx': 6, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': True}
    INFO INSTRUCTION: 25 | JIFNZ:NOT Z: DR => IP | AR: 6 | IP: 1 | DR: 1 | ACC: o | NZC: 0 | TICK: 108| CR:{'idx': 7, 'opcode': 'jifnz', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 26 | STORE: ACC => RAM[AR] | AR: 1 | IP: 2 | DR: 1 | ACC: o | NZC: 0 | TICK: 112| CR:{'idx': 1, 'opcode': 'store', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 27 | LOAD: DR => ACC | AR: 16 | IP: 3 | DR: 6 | ACC: 6 | NZC: 0 | TICK: 116| CR:{'idx': 2, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 28 | ADD: ACC + DR => ACC | AR: 3 | IP: 4 | DR: 1 | ACC: 7 | NZC: 0 | TICK: 120| CR:{'idx': 3, 'opcode': 'add', 'operand': 1, 'address': False, 'relative': False}
    INFO INSTRUCTION: 29 | STORE: ACC => RAM[AR] | AR: 16 | IP: 5 | DR: 1 | ACC: 7 | NZC: 0 | TICK: 124| CR:{'idx': 4, 'opcode': 'store', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 30 | LOAD: DR => ACC | AR: 16 | IP: 6 | DR: , | ACC: , | NZC: 0 | TICK: 130| CR:{'idx': 6, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': True}
    INFO INSTRUCTION: 31 | JIFNZ:NOT Z: DR => IP | AR: 6 | IP: 1 | DR: 1 | ACC: , | NZC: 0 | TICK: 134| CR:{'idx': 7, 'opcode': 'jifnz', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 32 | STORE: ACC => RAM[AR] | AR: 1 | IP: 2 | DR: 1 | ACC: , | NZC: 0 | TICK: 138| CR:{'idx': 1, 'opcode': 'store', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 33 | LOAD: DR => ACC | AR: 16 | IP: 3 | DR: 7 | ACC: 7 | NZC: 0 | TICK: 142| CR:{'idx': 2, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 34 | ADD: ACC + DR => ACC | AR: 3 | IP: 4 | DR: 1 | ACC: 8 | NZC: 0 | TICK: 146| CR:{'idx': 3, 'opcode': 'add', 'operand': 1, 'address': False, 'relative': False}
    INFO INSTRUCTION: 35 | STORE: ACC => RAM[AR] | AR: 16 | IP: 5 | DR: 1 | ACC: 8 | NZC: 0 | TICK: 150| CR:{'idx': 4, 'opcode': 'store', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 36 | LOAD: DR => ACC | AR: 16 | IP: 6 | DR:   | ACC:   | NZC: 0 | TICK: 156| CR:{'idx': 6, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': True}
    INFO INSTRUCTION: 37 | JIFNZ:NOT Z: DR => IP | AR: 6 | IP: 1 | DR: 1 | ACC:   | NZC: 0 | TICK: 160| CR:{'idx': 7, 'opcode': 'jifnz', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 38 | STORE: ACC => RAM[AR] | AR: 1 | IP: 2 | DR: 1 | ACC:   | NZC: 0 | TICK: 164| CR:{'idx': 1, 'opcode': 'store', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 39 | LOAD: DR => ACC | AR: 16 | IP: 3 | DR: 8 | ACC: 8 | NZC: 0 | TICK: 168| CR:{'idx': 2, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 40 | ADD: ACC + DR => ACC | AR: 3 | IP: 4 | DR: 1 | ACC: 9 | NZC: 0 | TICK: 172| CR:{'idx': 3, 'opcode': 'add', 'operand': 1, 'address': False, 'relative': False}
    INFO INSTRUCTION: 41 | STORE: ACC => RAM[AR] | AR: 16 | IP: 5 | DR: 1 | ACC: 9 | NZC: 0 | TICK: 176| CR:{'idx': 4, 'opcode': 'store', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 42 | LOAD: DR => ACC | AR: 16 | IP: 6 | DR: w | ACC: w | NZC: 0 | TICK: 182| CR:{'idx': 6, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': True}
    INFO INSTRUCTION: 43 | JIFNZ:NOT Z: DR => IP | AR: 6 | IP: 1 | DR: 1 | ACC: w | NZC: 0 | TICK: 186| CR:{'idx': 7, 'opcode': 'jifnz', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 44 | STORE: ACC => RAM[AR] | AR: 1 | IP: 2 | DR: 1 | ACC: w | NZC: 0 | TICK: 190| CR:{'idx': 1, 'opcode': 'store', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 45 | LOAD: DR => ACC | AR: 16 | IP: 3 | DR: 9 | ACC: 9 | NZC: 0 | TICK: 194| CR:{'idx': 2, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 46 | ADD: ACC + DR => ACC | AR: 3 | IP: 4 | DR: 1 | ACC: 10 | NZC: 0 | TICK: 198| CR:{'idx': 3, 'opcode': 'add', 'operand': 1, 'address': False, 'relative': False}
    INFO INSTRUCTION: 47 | STORE: ACC => RAM[AR] | AR: 16 | IP: 5 | DR: 1 | ACC: 10 | NZC: 0 | TICK: 202| CR:{'idx': 4, 'opcode': 'store', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 48 | LOAD: DR => ACC | AR: 16 | IP: 6 | DR: o | ACC: o | NZC: 0 | TICK: 208| CR:{'idx': 6, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': True}
    INFO INSTRUCTION: 49 | JIFNZ:NOT Z: DR => IP | AR: 6 | IP: 1 | DR: 1 | ACC: o | NZC: 0 | TICK: 212| CR:{'idx': 7, 'opcode': 'jifnz', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 50 | STORE: ACC => RAM[AR] | AR: 1 | IP: 2 | DR: 1 | ACC: o | NZC: 0 | TICK: 216| CR:{'idx': 1, 'opcode': 'store', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 51 | LOAD: DR => ACC | AR: 16 | IP: 3 | DR: 10 | ACC: 10 | NZC: 0 | TICK: 220| CR:{'idx': 2, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 52 | ADD: ACC + DR => ACC | AR: 3 | IP: 4 | DR: 1 | ACC: 11 | NZC: 0 | TICK: 224| CR:{'idx': 3, 'opcode': 'add', 'operand': 1, 'address': False, 'relative': False}
    INFO INSTRUCTION: 53 | STORE: ACC => RAM[AR] | AR: 16 | IP: 5 | DR: 1 | ACC: 11 | NZC: 0 | TICK: 228| CR:{'idx': 4, 'opcode': 'store', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 54 | LOAD: DR => ACC | AR: 16 | IP: 6 | DR: r | ACC: r | NZC: 0 | TICK: 234| CR:{'idx': 6, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': True}
    INFO INSTRUCTION: 55 | JIFNZ:NOT Z: DR => IP | AR: 6 | IP: 1 | DR: 1 | ACC: r | NZC: 0 | TICK: 238| CR:{'idx': 7, 'opcode': 'jifnz', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 56 | STORE: ACC => RAM[AR] | AR: 1 | IP: 2 | DR: 1 | ACC: r | NZC: 0 | TICK: 242| CR:{'idx': 1, 'opcode': 'store', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 57 | LOAD: DR => ACC | AR: 16 | IP: 3 | DR: 11 | ACC: 11 | NZC: 0 | TICK: 246| CR:{'idx': 2, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 58 | ADD: ACC + DR => ACC | AR: 3 | IP: 4 | DR: 1 | ACC: 12 | NZC: 0 | TICK: 250| CR:{'idx': 3, 'opcode': 'add', 'operand': 1, 'address': False, 'relative': False}
    INFO INSTRUCTION: 59 | STORE: ACC => RAM[AR] | AR: 16 | IP: 5 | DR: 1 | ACC: 12 | NZC: 0 | TICK: 254| CR:{'idx': 4, 'opcode': 'store', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 60 | LOAD: DR => ACC | AR: 16 | IP: 6 | DR: l | ACC: l | NZC: 0 | TICK: 260| CR:{'idx': 6, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': True}
    INFO INSTRUCTION: 61 | JIFNZ:NOT Z: DR => IP | AR: 6 | IP: 1 | DR: 1 | ACC: l | NZC: 0 | TICK: 264| CR:{'idx': 7, 'opcode': 'jifnz', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 62 | STORE: ACC => RAM[AR] | AR: 1 | IP: 2 | DR: 1 | ACC: l | NZC: 0 | TICK: 268| CR:{'idx': 1, 'opcode': 'store', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 63 | LOAD: DR => ACC | AR: 16 | IP: 3 | DR: 12 | ACC: 12 | NZC: 0 | TICK: 272| CR:{'idx': 2, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 64 | ADD: ACC + DR => ACC | AR: 3 | IP: 4 | DR: 1 | ACC: 13 | NZC: 0 | TICK: 276| CR:{'idx': 3, 'opcode': 'add', 'operand': 1, 'address': False, 'relative': False}
    INFO INSTRUCTION: 65 | STORE: ACC => RAM[AR] | AR: 16 | IP: 5 | DR: 1 | ACC: 13 | NZC: 0 | TICK: 280| CR:{'idx': 4, 'opcode': 'store', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 66 | LOAD: DR => ACC | AR: 16 | IP: 6 | DR: d | ACC: d | NZC: 0 | TICK: 286| CR:{'idx': 6, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': True}
    INFO INSTRUCTION: 67 | JIFNZ:NOT Z: DR => IP | AR: 6 | IP: 1 | DR: 1 | ACC: d | NZC: 0 | TICK: 290| CR:{'idx': 7, 'opcode': 'jifnz', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 68 | STORE: ACC => RAM[AR] | AR: 1 | IP: 2 | DR: 1 | ACC: d | NZC: 0 | TICK: 294| CR:{'idx': 1, 'opcode': 'store', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 69 | LOAD: DR => ACC | AR: 16 | IP: 3 | DR: 13 | ACC: 13 | NZC: 0 | TICK: 298| CR:{'idx': 2, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 70 | ADD: ACC + DR => ACC | AR: 3 | IP: 4 | DR: 1 | ACC: 14 | NZC: 0 | TICK: 302| CR:{'idx': 3, 'opcode': 'add', 'operand': 1, 'address': False, 'relative': False}
    INFO INSTRUCTION: 71 | STORE: ACC => RAM[AR] | AR: 16 | IP: 5 | DR: 1 | ACC: 14 | NZC: 0 | TICK: 306| CR:{'idx': 4, 'opcode': 'store', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 72 | LOAD: DR => ACC | AR: 16 | IP: 6 | DR: ! | ACC: ! | NZC: 0 | TICK: 312| CR:{'idx': 6, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': True}
    INFO INSTRUCTION: 73 | JIFNZ:NOT Z: DR => IP | AR: 6 | IP: 1 | DR: 1 | ACC: ! | NZC: 0 | TICK: 316| CR:{'idx': 7, 'opcode': 'jifnz', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 74 | STORE: ACC => RAM[AR] | AR: 1 | IP: 2 | DR: 1 | ACC: ! | NZC: 0 | TICK: 320| CR:{'idx': 1, 'opcode': 'store', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 75 | LOAD: DR => ACC | AR: 16 | IP: 3 | DR: 14 | ACC: 14 | NZC: 0 | TICK: 324| CR:{'idx': 2, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 76 | ADD: ACC + DR => ACC | AR: 3 | IP: 4 | DR: 1 | ACC: 15 | NZC: 0 | TICK: 328| CR:{'idx': 3, 'opcode': 'add', 'operand': 1, 'address': False, 'relative': False}
    INFO INSTRUCTION: 77 | STORE: ACC => RAM[AR] | AR: 16 | IP: 5 | DR: 1 | ACC: 15 | NZC: 0 | TICK: 332| CR:{'idx': 4, 'opcode': 'store', 'operand': 16, 'address': True, 'relative': False}
    INFO INSTRUCTION: 78 | LOAD: DR => ACC | AR: 16 | IP: 6 | DR:  | ACC:  | NZC: 10 | TICK: 338| CR:{'idx': 6, 'opcode': 'load', 'operand': 16, 'address': True, 'relative': True}
    INFO INSTRUCTION: 79 | JIFNZ:NOT Z: DR => IP | AR: 6 | IP: 7 | DR:  | ACC:  | NZC: 10 | TICK: 342| CR:{'idx': 7, 'opcode': 'jifnz', 'operand': 1, 'address': True, 'relative': False}
    INFO INSTRUCTION: 80 | HLT | AR: 7 | IP: 8 | DR:  | ACC:  | NZC: 10 | TICK: 343| CR:{'idx': 8, 'opcode': 'hlt', 'operand': None, 'address': False, 'relative': False}
    INFO ['H', 'e', 'l', 'l', 'o', ',', ' ', 'w', 'o', 'r', 'l', 'd', '!']
